src/pybind/mgr/dashboard/frontend/src/app/core/auth/user-form/user-form.component.spec.ts

```diff
@@ -0,0 +1,243 @@
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { Component } from '@angular/core';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { ReactiveFormsModule } from '@angular/forms';
+import { Router, Routes } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+
+import { ToastModule } from 'ng2-toastr';
+import { BsModalService } from 'ngx-bootstrap';
+import { of } from 'rxjs';
+
+import { RoleService } from '../../../shared/api/role.service';
+import { UserService } from '../../../shared/api/user.service';
+import { ComponentsModule } from '../../../shared/components/components.module';
+import { CdFormGroup } from '../../../shared/forms/cd-form-group';
+import { AuthStorageService } from '../../../shared/services/auth-storage.service';
+import { NotificationService } from '../../../shared/services/notification.service';
+import { SharedModule } from '../../../shared/shared.module';
+import { configureTestBed } from '../../../shared/unit-test-helper';
+import { UserFormComponent } from './user-form.component';
+import { UserFormModel } from './user-form.model';
+
+describe('UserFormComponent', () => {
+  let component: UserFormComponent;
+  let form: CdFormGroup;
+  let fixture: ComponentFixture<UserFormComponent>;
+  let httpTesting: HttpTestingController;
+  let userService: UserService;
+  let modalService: BsModalService;
+  let router: Router;
+  const setUrl = (url) => Object.defineProperty(router, 'url', { value: url });
+
+  @Component({ selector: 'cd-fake', template: '' })
+  class FakeComponent {}
+
+  const routes: Routes = [
+    { path: 'login', component: FakeComponent },
+    { path: 'users', component: FakeComponent }
+  ];
+
+  configureTestBed({
+    imports: [
+      [RouterTestingModule.withRoutes(routes)],
+      HttpClientTestingModule,
+      ReactiveFormsModule,
+      RouterTestingModule,
+      ComponentsModule,
+      ToastModule.forRoot(),
+      SharedModule
+    ],
+    declarations: [UserFormComponent, FakeComponent]
+  });
```

Comment:
Change it to `}, true);` as the `edit mode` tests will fail, except the first, without a reset of all loaded modules.

Reply:
Done